app/main/views/search.py

```diff
@@ -59,6 +59,20 @@ def create(index_name):
 
 @main.route('/<string:index_name>', methods=['DELETE'])
 def delete(index_name):
+    status, status_code = status_for_index(index_name)
+
+    if status_code != 200:
```

Comment:
Is this just checking that the index exists before going any further?  Are there times when the status might not be 200 (if the index is broken somehow? Does that still give a 200?), but where we'd still want to go ahead with the delete?